server/db/mongo/modules/settingsModule.js

```
@@ -14,12 +14,22 @@ const getAppSettings = async () => {
 	try {
-		console.log(newSettings);
-		const settings = await AppSettings.findOneAndUpdate(
-			{},
-			{ $set: newSettings },
-			{ new: true, upsert: true }
-		);
+		const update = { $set: { ...newSettings } };
+
+		if (newSettings.pagespeedApiKey === "") {
+			update.$unset = { pagespeedApiKey: "" };
+			delete update.$set.pagespeedApiKey;
```

Biome (1.9.4)

[error] Avoid the delete operator which can impact performance.

Unsafe fix: Use an undefined assignment instead.


(lint/performance/noDelete)